src/sedcli_main.c

```diff
@@ -736,6 +783,55 @@ static int handle_mbr_control(void)
         return ret;
 }
 
+static int handle_write_mbr(void)
+{
+	struct sed_device *dev = NULL;
+	int ret, mbr_fd;
+	struct stat mbr_st;
+	void *mbr_mmap;
+
+	sedcli_printf(LOG_INFO, "Enter Admin1 password: ");
+
+	ret = get_password((char *) opts->pwd.key, &opts->pwd.len,
+			SED_MIN_KEY_LEN, SED_MAX_KEY_LEN);
+	if (ret)
+		return -1;
+
+	mbr_fd = open(opts->file_path, O_RDONLY | O_CLOEXEC);
+	if (fstat(mbr_fd, &mbr_st) == -1) {
```

Comment:
fstat checks could be done before prompting user for password. this would offer better experience